monitoring/src/check-page/consent-or-pay-tcfv2.ts

```diff
@@ -0,0 +1,599 @@
+				BannerInteractions.REJECT_ALL,
+			);
+			await checkCMPIsNotVisible(page);
+			await isUsingNonPersonalisedAds(page);
+			break;
+	}
+
+	if (nextUrl) {
+		await checkSubsequentPage(
+			context,
+			config,
+			nextUrl,
+			BannerType.NON_ADVERTISING,
+		);
+	}
+
+	await page.close();
+	await browser.close();
+	log_info('Checking non advertising banner: Complete');
+
+};
+
+/**
+ * Checks that ads load correctly for the second page a user goes to
+ * when visiting the site, with respect to and interaction with the CMP.
+ *
+ * @param {BrowserContext} context
+ * @param {Config} config
+ * @param {string} url
+ */
+const checkSubsequentPage = async (
+	context: BrowserContext,
+	config: Config,
+	url: string,
+	bannerType: Banner,
+) => {
+	log_info(`Start checking subsequent Page URL: ${url}`);
+	const page: Page = await makeNewPage(context);
+	await loadPage(page, url);
+	// There is no CMP since this we have already accepted this on a previous page.
+	await checkCMPIsNotVisible(page);
+
+	switch (bannerType) {
+		case BannerType.CONSENT_OR_PAY_SIGNED_IN:
+			await isUsingPersonalisedAds(page);
+			break;
+		case BannerType.CONSENT_OR_PAY_SIGNED_OUT:
+			await isUsingPersonalisedAds(page);
+			break;
+		case BannerType.NON_ADVERTISING:
+			await isUsingNonPersonalisedAds(page);
+			break;
+	}
+
+	await Promise.all([clearCookies(page), clearLocalStorage(page)]);
+	await reloadPage(page);
+	await checkTopAdDidNotLoad(page);
+	await clickAcceptAllCookies(config, page, "Yes I'm Happy");
+	await Promise.all([
+		checkCMPIsNotVisible(page),
+		isUsingPersonalisedAds(page),
+	]);
+	await page.close();
+	log_info(`Checking subsequent Page URL: ${url} Complete`);
+};
+
+/**
+ * This is the function tests the page for two different scenarios:
+ * 1. A user visits the home page (uses the frontend repo) and navigates
+ * to an article page (uses dotcom-rendering)
+ * 2. A user visit only the article page
+ *
+ * @param {Config} config
+ * @return {*}  {Promise<void>}
+ */
+export const mainCheck = async function (config: Config): Promise<void> {
+	await checkConsentOrPayBanner(
+		config,
+		`${config.frontUrl}?adtest=fixed-puppies`,
+		`${config.articleUrl}?adtest=fixed-puppies`,
+		BannerInteractions.ACCEPT_ALL,
+		BannerType.CONSENT_OR_PAY_SIGNED_OUT,
+	);
+	log_line();
+
+	await checkConsentOrPayBanner(
+		config,
+		`${config.frontUrl}?adtest=fixed-puppies`,
+		``,
+		BannerInteractions.REJECT_AND_SUBSCRIBE,
+		BannerType.CONSENT_OR_PAY_SIGNED_OUT,
+	);
+	log_line();
+
+	// await checkConsentOrPayBanner(
+	// 	config,
+	// 	`${config.frontUrl}?adtest=fixed-puppies`,
+	// 	`${config.articleUrl}?adtest=fixed-puppies`,
+	// 	BannerInteractions.ACCEPT_ALL,
+	// 	BannerType.CONSENT_OR_PAY_SIGNED_IN,
+	// );
+	// log_line();
+
+	// await checkConsentOrPayBanner(
+	// 	config,
+	// 	`${config.articleUrl}?adtest=fixed-puppies`,
+	// 	``,
+	// 	BannerInteractions.REJECT_AND_SUBSCRIBE,
+	// 	BannerType.CONSENT_OR_PAY_SIGNED_IN,
+	// );
+	// log_line();
```

Comment:
Should the signed in tests be commented out?

Reply:
Currently, I'm unable to mock a signed in user. I was going to investigate this in another pr.